tests/test_analysis/test_feature_relevance/test_relevance_table.py

```diff
@@ -34,3 +36,110 @@ def test_model_relevance_table(simple_df_relevance):
     assert np.allclose(relevance_table["regressor_1"]["1"], 1)
     assert np.allclose(relevance_table["regressor_2"]["1"], 0)
     assert relevance_table["regressor_1"]["2"] < relevance_table["regressor_2"]["2"]
+
+
+@pytest.fixture()
+def exog_and_target_dfs():
+    seg = ["a"] * 30 + ["b"] * 30
+    time = list(pd.date_range("2020-01-01", "2021-01-01")[:30])
+    timestamps = time * 2
+    target = np.arange(60)
+    df = pd.DataFrame({"segment": seg, "timestamp": timestamps, "target": target})
+    ts = TSDataset.to_dataset(df)
+
+    cast = ["1.1"] * 10 + ["2"] * 10 + ["56.1"] * 10
+    no_cast = ["1.1"] * 10 + ["two"] * 10 + ["56.1"] * 10
+    none = [1] * 10 + [2] * 10 + [56.1] * 10
+    none[10] = None
+    df = pd.DataFrame(
+        {
+            "timestamp": time,
+            "exog1": np.arange(100, 70, -1),
+            "exog2": np.sin(np.arange(30) / 10),
+            "exog3": np.exp(np.arange(30)),
+            "cast": cast,
+            "no_cast": no_cast,
+            "none": none,
+        }
+    )
+    df["cast"] = df["cast"].astype("category")
+    df["no_cast"] = df["no_cast"].astype("category")
+    df_exog = duplicate_data(df, segments=["a", "b"])
+    return ts, df_exog
+
+
+@pytest.mark.parametrize(
+    "columns,match",
+    (
+        (["exog1", "exog2", "exog3", "cast"], "Exogenous data contains columns with category type"),
+        (["exog1", "exog2", "exog3", "none"], "Exogenous or target data contains None"),
+    ),
+)
+def test_warnings_statistic_table(columns, match, exog_and_target_dfs):
+    df, df_exog = exog_and_target_dfs
+    df_exog = df_exog[[i for i in df_exog.columns if i[1] in columns]]
+    with pytest.warns(UserWarning, match=match):
+        get_statistics_relevance_table(df=df, df_exog=df_exog)
+
+
+def test_errors_statistic_table(exog_and_target_dfs):
+    df, df_exog = exog_and_target_dfs
+    with pytest.raises(ValueError, match="column cannot be cast to float type!"):
+        get_statistics_relevance_table(df=df, df_exog=df_exog)
+
+
+def test_work_statistic_table(exog_and_target_dfs):
+    df, df_exog = exog_and_target_dfs
+    df_exog = df_exog[[i for i in df_exog.columns if i[1] != "no_cast"]]
+    get_statistics_relevance_table(df=df, df_exog=df_exog)
+
+
+def test_target_none_statistic_table(exog_and_target_dfs):
+    df, df_exog = exog_and_target_dfs
+    tmp = np.arange(len(df["a", "target"]), dtype=float)
+    tmp[5] = np.nan
+    df["a", "target"] = tmp
+
+    df_exog = df_exog[[i for i in df_exog.columns if i[1][:-1] == "exog"]]
+    with pytest.warns(UserWarning, match="Exogenous or target data contains None"):
+        get_statistics_relevance_table(df=df, df_exog=df_exog)
+
+
+def test_target_none_model_table(exog_and_target_dfs):
+    df, df_exog = exog_and_target_dfs
+    tmp = np.arange(len(df["a", "target"]), dtype=float)
```

Comment:
Let'c create the separate fixture for this case, you can make it out of the exog_and_target_dfs